pkg/util/util.go

```diff
@@ -113,6 +113,38 @@ func UnArchiveTar(reader io.Reader, destDir string, arg ...string) error {
 	return nil
 }
 
+// ExrtactSingleDirTar unarchives a tar file and streams its files
+// using the specified io.Reader to the specified destination.
+func ExrtactSingleDirTar(reader io.Reader, destDir string, extractDir string) error {
```

Comment:
maybe we could modify 'UnArchiveTar' function and reuse it here, instead of copy the logic. could be done later on..

Reply:
There is also a typo in the name it should be ExtractSingleDirTar